pcap.c

```diff
@@ -78,6 +78,10 @@ struct rtentry;		/* declarations in <net/if.h> */
 #define INT_MAX		2147483647
 #endif
 
+#if defined(_MSC_VER) && !defined(strdup)
```

Comment:
And again...

Reply:
Corrects the warning.

Reply:
Again, why must this be handled in several places in the source rather than by a change to `portability.h`?

Reply:
I figured it was a more reliable way to correct the issue, even if it is uglier I agree. The potential problem with making the change in portability.h is that something included after it may try to redefine it (and this may be a standard header file, apparently). Doing this reliably in portability.h would then require ensuring the file is included as close to the source code as possible, which is more intrusive and probably not a good idea. Defining the macro directly in the source file after all headers have been included seemed like the safest, most conservative option.

Reply:
Making the change in source files runs the risk of new code being written, using `strdup()`, but not adding the #define (lots of new stuff is written on UN\*X, where the #define isn't necessary).